### ISSUE TITLE
React: Issues Mega menu options navigation

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#4196]
## Detailed description

Issue: The options within the mentioned menus are navigable with tab/shift tab keys instead of Up/Down Arrow keys. This behaviour does not adhere to the aria practices.

Expected: 
Either these menus should be made in adherence to ARIA standards. Or an accessible alternate should be given. 
Below is the reference link for aria 1.1 examples. Note: We recommend the second approach. 
1. Navigation Menu bar Example | WAI-ARIA Authoring Practices 1.1
https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html
2. Example Disclosure for Navigation Menus | WAI-ARIA Authoring Practices 1.1
https://www.w3.org/TR/wai-aria-practices/examples/disclosure/disclosure-navigation.html

**Comment by annawen1** [comment]
Looking at the second approach listed and recommended, it does seem like the up and down arrow key navigation is optional according to the aria practices link.

Example Disclosure for Navigation Menus | WAI-ARIA Authoring Practices 1.1
https://www.w3.org/TR/wai-aria-practices/examples/disclosure/disclosure-navigation.html

**Comment by asudoh** [comment]
Right - I think we can basically choose one vs. another, and Carbon core UI shell takes the tab key approach. Another pattern is useful when there are lots of nav items in the nav group; e.g. When there are lots of radio buttons in a radio group and user wants to navigate from radio group to what comes after the radio group, having to hit multiple tab keys within radio box is cumbersome, and [here](https://developers.google.com/web/fundamentals/accessibility/focus/using-tabindex#managing_focus_in_components) is the pattern to mitigate such problem. However in mega menu case, I think user can simply hit <kbd>ESC</kbd> key to go out of mega menu.

**Comment by annawen1** [comment]
After conversation with QA team, it seems what is needed is to remove the "menu" role from the megamenu
